Assets/HoloToolkit/Sharing/Scripts/Utilities/ManualIpConfiguration.cs

```diff
@@ -0,0 +1,204 @@
+﻿using System;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace HoloToolkit.Sharing.Utilities
+{
+    /// <summary>
+    /// Utility for connecting to Sharing Service by Ip Address from inside application at runtime.
+    /// </summary>
+    public class ManualIpConfiguration : MonoBehaviour
+    {
+        public string IpAddress { get { return ipAddress.text; } }
+
+        /// <summary>
+        /// Hides the UI when connection is made.
+        /// </summary>
+        [Tooltip("Hides the UI when connection is made.")]
+        public bool HideWhenConnected;
```

Comment:
Might be worth adding a voice command or something to bring the UI up again just in case.